bin/build.sh

```diff
@@ -1,73 +1,78 @@
 wca_docs_url="https://documents.worldcubeassociation.org/"
+# Absolute paths to the logo file, the main stylesheet and the edudoc header
 logo=$(realpath "./assets/WCAlogo_notext.svg")
-# Absolute paths to the stylesheets and the edudoc header
-stylesheet=$(realpath "./assets/style.css")
-edudoc_stylesheet=$(realpath "./assets/edudoc-style.css")
+main_stylesheet=./assets/style.css
+main_stylesheet_path=$(realpath "$main_stylesheet")
 edudoc_header=$(realpath "./assets/edudoc-header.html")
 # Current date
 compile_date=$(date '+%Y-%m-%d')
 
-# Function for converting Markdown files from a given folder into PDF files using the given stylesheet
-# The first argument ($1) is the folder and the second argument is the stylesheet ($2)
+# Function for converting Markdown files from a given directory into PDF files.
+# The first argument ($1) refers to the directory name (e.g: documents or edudoc).
 convert_to_pdf() {
+  rm -rf "build/$1"
   cp -r "$1/" build/
 
-  # Find formal and legal Markdown files and build PDFs out of them.
+  # Find Markdown files and build PDFs out of them.
   find "build/$1" -name '*.md' | while read file; do
-    echo "Processing $file using the ${1} stylesheet"
+    pdf_path="${file%.md}.pdf"
+    html_path="${file%.md}.html"
+    # Use first header as the document title; remove all # characters and whitespace from the beginning
+    document_title=$(head -n 1 "$file" | sed -E "s/^#+\s*//")
+    # Absolute path to the custom stylesheet
+    custom_stylesheet="./assets/$1-style.css"
+    custom_stylesheet_path=$(realpath "$custom_stylesheet")
 
-    pdf_name="${file%.md}.pdf"
-    html_name="${file%.md}.html"
-    
-    file_headline=$(head -n 1 "$file")
-    document_title=$(echo "$file_headline" | sed -E "s/#+\s*//")
+    echo "Converting ${file#build/} using stylesheets ${main_stylesheet#./assets/} and ${custom_stylesheet#./assets/}..."
 
     # Replace wca{...} and wcadoc{...} with absolute WCA URLs (the correct URLs only get inserted in production)
     sed -Ei "s#wca\{([^}]*)\}#$wca_url\1#g" "$file"
     sed -Ei "s#wcadoc\{([^}]*)\}#$wca_docs_url\1#g" "$file"
     # Replace {logo} with the path to the WCA logo in /assets
     sed -Ei "s#\{logo\}#$logo#g" "$file"
 
-    # This creates the custom header for each edudoc and saves it next to the doc that will use it
-    if [ $1 = "edudoc" ]; then
-      header_html="${file%.md}-header.html"
-    
+    # This creates the custom header for each edudoc and prepends it to each file
+    if [ "$1" = "edudoc" ]; then
       # Set the document title, path to the logo and the date
-      sed -E "s#DOCUMENT_TITLE#$document_title#g" "$edudoc_header" |
-      sed -E "s#WCA_LOGO_PATH#$logo#g" |
-      sed -E "s#DATE#$compile_date#g" > "$header_html"
+      sed -E "s#\{document_title\}#$document_title#g" "$edudoc_header" |
+      sed -E "s#\{wca_logo_path\}#$logo#g" |
+      sed -E "s#\{date\}#$compile_date#g" |
+      echo -e "$(cat /dev/stdin)$(cat $file)" > "$file"
```

Comment:
I still feel quite strongly about this `echo` with a wrapped `cat /dev/stdin` call.
What's so hurtful about a plain, simple, readable

```bash
sed -E | sed -E | sed -E > $target_file
cat $origin_file >> $target_file
```

Reply:
The problem is that we need to prepend here, not append. That's what the wrapped cat call achieves here.

Reply:
Found a different, hopefully cleaner, solution using tee

Reply:
That actually didn't work, and it breaks every other document. Reverting it to the one line echo solution.

Reply:
Let me know if you figure out a cleaner way to prepend the header html

Reply:
Uhm... Doesn't my original suggestion do just that?
Run some `sed`s, write them to a new, temporary file (i.e. they get added to the *beginning* of that new file) and then use `>>` to _append_ the original file to the `sed`'ed stuff.

Reply:
Ohh, so you meant to append to the new file that has the header html in it. I'm dumb, I'll do that then.

Reply:
Done